library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -2372,72 +2083,61 @@ impl<T, A: Allocator> VecDeque<T, A> {
     /// ```
     #[stable(feature = "deque_make_contiguous", since = "1.48.0")]
     pub fn make_contiguous(&mut self) -> &mut [T] {
+        if T::IS_ZST {
```

Comment:
hmm, is *this* the right place for an `IS_ZST` check?  I wonder if perhaps `is_contiguous` should just always say that the deque is contiguous when `T::IS_ZST` -- after all, `.add(self.head)` is a NOP for ZSTs.

Probably fine to leave it like this for this PR, though.

Reply:
Thinking more, that leads to the `slice::Iter`-style thing where it's a fairly different implementation for ZSTs basically everywhere -- leaving `head` alone and just tracking `len` because that's the only thing that matters -- but that's a huge change that definitely I wouldn't want to include in this PR.